### PR TITLE
add VerifiedMaintainer field to Addons and display a warning if empty

### LINKED ISSUE #14493: Add GitHub maintainer field to addons & warn for all addons missing a maintainer
We should add a GitHub maintainer field to the addons (below):
https://github.com/kubernetes/minikube/blob/044fe4c1fd2779b5ac3e4974556c3ff7b4d26792/pkg/minikube/assets/addons.go#L37-L47

An empty field should be added for each addon, and for each addon that has an empty field, when the user tries to enable the addon it should give a warning to the user that there's not a verified maintainer for the addon.

### PR DESCRIPTION
**Pull request by inifares23lab**
The Addon type now has another string Field: VerifiedMaintainer
The new field is added to the NewAddon function and to the Addons map.

When enabling an addon if the VerifiedMaintainer field is empty a warning message is displayed:

! The <ADDON_NAME> addon doesn't have a verified maintainer.
Since for now I just added an empty string for every addon, it will always display the warning when enabling

fixes #14493

### CODE REVIEW

**Comment by inifares23lab** [comment]
Hi @spowelljr
apologies
formatted and pushed now

**Comment by spowelljr** [comment]
Hi @inifares23lab, thanks for formatting the code.

Could you wrap all `Maintainers` that aren't `Google` or `Kubernetes` with `3rd party(X)`.

For example `Portainer.io` to `3rd party (Portainer.io)`

Could you also add `3rd party (Gluster)` as the maintainer for `storage-provisioner-gluster`.

If the `Maintainer` is `Kubernetes` or `Google` could you make the message be the following:
```
<addon_name> is an addon maintained by <maintainer> for any concerns contact minikube on GitHub.
You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS
```

For the rest of addons could you update your message to:
```
<addon_name> is A 3rd party addon and not maintained or verified by minikube maintainers, enable at your own risk.
```

And for 3rd party addons, if there's a verified maintainer for the addon could you also add the following line:
```
<addonn_name> is maintained by <maintainer> for any concerns contact <verified_maintaer> on GitHub.
```

**Comment by inifares23lab** [comment]
Hello

- updated messages and handled the 3 different cases
- updated 3rd party maintainers and fixed some typos

examples:
💡  auto-pause is an addon maintained by Google. For any concerns contact minikube on GitHub.
    You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS

❗  ambassador is A 3rd party addon and not maintained or verified by minikube maintainers, enable at your own risk.

💡  metallb is maintained by 3rd party (MetalLB) for any concerns contact MOCK-VERIFI
ED-MAINTAINER on GitHub.

File: cmd/minikube/cmd/config/enable.go [hunk 1]
```diff
@@ -52,6 +53,21 @@ var addonsEnableCmd = &cobra.Command{
 		if addon == "olm" {
 			out.Styled(style.Warning, "The OLM addon has stopped working, for more details visit: https://github.com/operator-framework/operator-lifecycle-manager/issues/2534")
 		}
+		addonBundle, ok := assets.Addons[addon]
+		if ok {
+			maintainer := addonBundle.Maintainer
+			if maintainer == "Google" || maintainer == "Kubernetes" {
+				out.Styled(style.Tip, `{{.addon}} is an addon maintained by {{.maintainer}}. For any concerns contact minikube on GitHub.
+    You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS`,
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/enable.go:61]:
```suggestion
You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS`,
```

File: cmd/minikube/cmd/config/enable.go [hunk 2]
```diff
@@ -52,6 +53,21 @@ var addonsEnableCmd = &cobra.Command{
...
+					out.V{"addon": addon, "maintainer": maintainer})
+			} else if verifiedMaintainer := addonBundle.VerifiedMaintainer; verifiedMaintainer != "" {
+				out.Styled(style.Tip, `{{.addon}} is maintained by {{.maintainer}} for any concerns contact {{.verified_maintainer}} on GitHub.`,
+					out.V{"addon": addon, "maintainer": maintainer, "verified_maintainer": verifiedMaintainer})
+			} else {
+				out.Styled(style.Warning, `{{.addon}} is A 3rd party addon and not maintained or verified by minikube maintainers, enable at your own risk.`,
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/enable.go:67]:
```suggestion
				out.Styled(style.Warning, `{{.addon}} is a 3rd party addon and not maintained or verified by minikube maintainers, enable at your own risk.`,
```

File: cmd/minikube/cmd/config/enable.go [hunk 3]
```diff
@@ -52,6 +53,21 @@ var addonsEnableCmd = &cobra.Command{
...
+		addonBundle, ok := assets.Addons[addon]
+		if ok {
+			maintainer := addonBundle.Maintainer
+			if maintainer == "Google" || maintainer == "Kubernetes" {
+				out.Styled(style.Tip, `{{.addon}} is an addon maintained by {{.maintainer}}. For any concerns contact minikube on GitHub.
+    You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS`,
+					out.V{"addon": addon, "maintainer": maintainer})
+			} else if verifiedMaintainer := addonBundle.VerifiedMaintainer; verifiedMaintainer != "" {
+				out.Styled(style.Tip, `{{.addon}} is maintained by {{.maintainer}} for any concerns contact {{.verified_maintainer}} on GitHub.`,
+					out.V{"addon": addon, "maintainer": maintainer, "verified_maintainer": verifiedMaintainer})
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/enable.go:65]:
```suggestion
					out.V{"addon": addon, "maintainer": maintainer, "verifiedMaintainer": verifiedMaintainer})
					
```
Can we keep normal Go casing here

**Review by spowelljr — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/config/enable.go (hunk 3 above)
Comment (inifares23lab) [reply cmd/minikube/cmd/config/enable.go]: of course

File: cmd/minikube/cmd/config/enable.go [hunk 4]
```diff
@@ -52,6 +53,21 @@ var addonsEnableCmd = &cobra.Command{
...
+		addonBundle, ok := assets.Addons[addon]
+		if ok {
+			maintainer := addonBundle.Maintainer
+			if maintainer == "Google" || maintainer == "Kubernetes" {
+				out.Styled(style.Tip, `{{.addon}} is an addon maintained by {{.maintainer}}. For any concerns contact minikube on GitHub.
+You can view the list of minikube maintainers at: https://github.com/kubernetes/minikube/blob/master/OWNERS`,
+					out.V{"addon": addon, "maintainer": maintainer})
+			} else if verifiedMaintainer := addonBundle.VerifiedMaintainer; verifiedMaintainer != "" {
+				out.Styled(style.Tip, `{{.addon}} is maintained by {{.maintainer}} for any concerns contact {{.verifiedMaintainer}} on GitHub.`,
+					out.V{"addon": addon, "maintainer": maintainer, "verifiedMaintainer": verifiedMaintainer})
```
Comment (spowelljr) [inline cmd/minikube/cmd/config/enable.go:65]:
```suggestion
			} else if addonBundle.VerifiedMaintainer != "" {
				out.Styled(style.Tip, `{{.addon}} is maintained by {{.maintainer}} for any concerns contact {{.verifiedMaintainer}} on GitHub.`,
					out.V{"addon": addon, "maintainer": maintainer, "verifiedMaintainer": addonBundle.VerifiedMaintainer})
```

The assigning within the if statement is a bit long and makes it a bit unreadable, I feel doing a direct comparison without assigning a new variable would be better

**Review by spowelljr — CHANGES_REQUESTED** [review]
The Google/Kubernetes and no verified maintainer case is perfect.

But if it's a 3rd party and has a verified maintainer it should have both messages:
```
<addon_name> is A 3rd party addon and not maintained or verified by minikube maintainers, enable at your own risk.
<addon_name> is maintained by <maintainer> for any concerns contact <verified_maintaer> on GitHub.
```

That ways they're notified it's a 3rd party addon and to enable at their own risk, but also pointed to someone if they have troubles.

**Comment by inifares23lab** [comment]
Hi @spowelljr

I agree that displaying both messages is more clear for the user, however I have one concern:

- the first message also states that the addon is not verified by minikube maintainers even if in some cases it has a VerifiedMaintainer


It may be better to further customize the messages for the two cases:

1. 3rd party without verified maintainer(same as before):
> <addon_name> is A 3rd party addon and not maintained or verified by minikube maintainers, enable at your own risk.

2. 3rd party with a verified maintainer(add the first message but omit "or verified"):
> <addon_name> is A 3rd party addon and not maintained by minikube maintainers, enable at your own risk.
> <addon_name> is maintained by <maintainer> for any concerns contact <verified_maintaer> on GitHub.


Let me know if it makes sense to you
thanks

**Comment by spowelljr** [comment]
Hi @inifares23lab, I know in the issue I originally mentioned a verified maintainer but that was just something I quickly jotted down and should have named it something else.

The field can be called anything, `GitHubMaintainer` or something like that, there's really no verifying the user themselves. Verified is related to the addon itself though, and any addon that's not maintained by Kubernetes or Google isn't verified.

So the message:
`<addon_name> is A 3rd party addon and not maintained or verified by minikube maintainers, enable at your own risk.`
Seems fine to me for both cases because there is no verified maintainer.

**Comment by inifares23lab** [comment]
understood, thank you for clarifying.
I applied the requested changes

File: pkg/minikube/assets/addons.go [hunk 5]
```diff
@@ -440,7 +442,7 @@ var Addons = map[string]*Addon{
 			vmpath.GuestAddonsDir,
 			"logviewer-rbac.yaml",
 			"0640"),
-	}, false, "logviewer", "", "", map[string]string{
+	}, false, "logviewer", "3rd party (ivans3)", "", "", map[string]string{
```
Comment (spowelljr) [inline pkg/minikube/assets/addons.go:445]:
```suggestion
	}, false, "logviewer", "3rd party (unknown)", "", "", map[string]string{
```

File: pkg/minikube/assets/addons.go [hunk 6]
```diff
@@ -145,21 +147,21 @@ var Addons = map[string]*Addon{
...
-	}, true, "default-storageclass", "Kubernetes", "https://minikube.sigs.k8s.io/docs/handbook/persistent_volumes/", nil, nil),
+	}, true, "default-storageclass", "Kubernetes", "", "https://minikube.sigs.k8s.io/docs/handbook/persistent_volumes/", nil, nil),
 	"pod-security-policy": NewAddon([]*BinAsset{
 		MustBinAsset(addons.PodSecurityPolicyAssets,
 			"pod-security-policy/pod-security-policy.yaml.tmpl",
 			vmpath.GuestAddonsDir,
 			"pod-security-policy.yaml",
 			"0640"),
-	}, false, "pod-security-policy", "", "", nil, nil),
+	}, false, "pod-security-policy", "3rd party (pod-security-policy)", "", "", nil, nil),
```
Comment (spowelljr) [inline pkg/minikube/assets/addons.go:157]:
```suggestion
	}, false, "pod-security-policy", "3rd party (unknown)", "", "", nil, nil),
```

File: pkg/minikube/assets/addons.go [hunk 7]
```diff
@@ -386,7 +388,7 @@ var Addons = map[string]*Addon{
 			vmpath.GuestAddonsDir,
 			"patch-coredns-job.yaml",
 			"0640"),
-	}, false, "registry-aliases", "", "", map[string]string{
+	}, false, "registry-aliases", "3rd party (registry-aliases)", "", "", map[string]string{
```
Comment (spowelljr) [inline pkg/minikube/assets/addons.go:391]:
```suggestion
	}, false, "registry-aliases", "3rd party (unknown)", "", "", map[string]string{
```

**Review by spowelljr — CHANGES_REQUESTED** [review]
Looks good! Just a few last comments

**Review by spowelljr — APPROVED** [review]
Looks good, thanks for your work on this!

**Pull request merged**